src/components/ContactCard/ContactCard.jsx

```diff
@@ -0,0 +1,84 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { translate } from "cozy-ui/react/I18n";
+import ContactIdentity from "./ContactIdentity";
+import ContactFields from "./ContactFields";
+import contactPropTypes from "../ContactPropTypes";
+
+const HeaderActions = () => null;
+
+export const supportedFieldsInOrder = [
+  "phone",
+  "email",
+  "address",
+  "cozy",
+  "company",
+  "birthday",
+  "note"
+];
+
+export const getFieldListFrom = contact =>
+  Object.keys(contact).map(type => ({ type, values: contact[type] }));
+export const filterFieldList = fields =>
+  fields.filter(
+    field => ["name", "_id", "_rev"].includes(field.type) === false
+  );
+export const groupUnsupportedFields = (fields, supportedFieldTypes) => {
+  const supportedFields = fields.filter(field =>
+    supportedFieldTypes.includes(field.type)
+  );
+  const unsupportedFields = fields.filter(
+    field => !supportedFieldTypes.includes(field.type)
+  );
+
+  return supportedFields.concat([
+    {
+      type: "other",
+      values: unsupportedFields.map(unsupportedField => unsupportedField.values)
+    }
+  ]);
+};
+export const orderFieldList = (fields, fieldsInOrder) =>
+  fields.sort(
+    (a, b) => fieldsInOrder.indexOf(a.type) - fieldsInOrder.indexOf(b.type)
+  );
+export const makeValuesArray = fields =>
+  fields.map(field => ({
+    ...field,
+    values: Array.isArray(field.values) ? field.values : [field.values]
+  }));
+
+const ContactCard = ({ t, contact }) => {
+  const fields = getFieldListFrom(contact);
+  const filteredFields = filterFieldList(fields);
+  const groupedFields = groupUnsupportedFields(
+    filteredFields,
+    supportedFieldsInOrder
+  );
+  const orderedFields = orderFieldList(groupedFields, supportedFieldsInOrder);
+  const normalizedFields = makeValuesArray(orderedFields);
```

Comment:
Yeah I didn't think of using that, but that's good. 
I really didn't want to write something like  `makeValuesArray(sort(groupFields(fieldsAsList(filterFields(contact)))))` which is perfectly unreadable, so using intermediary was my way of making things explicit.
Can't find the link right now, but apparently there's a flow operator coming to js which would be cool:

```
contact
|> filterFields
|> groupUnsupported
|> etc
```

Reply:
That is to avoid such discussion about implementation details that I suggest to expose an unique and well thought interface. Thus you can do whatever you like or want, whether you like `lodash/fp` functional programming implementation, or a more imperative list of intermediary values and computation, without changing anything about the consuming code or the awaited behavior (if the interface is covered by tests).

---

C'est pour éviter ce genre de discussion au sujet des détails d'implémentation que je propose toujours d'exposer une unique interface bien réfléchie. Ainsi tu peux te faire plaisir, que ce soit avec de la programmation fonctionnelle avec `lodash/fp`, ou avec une version plus impérative à base de variables et calculs intermédiaires, sans changer le code qui consomme l'API ou le comportement attendu (si l'interface est couverte par des tests).